source/common/network/io_socket_handle_impl.cc

```diff
@@ -131,5 +133,170 @@ IoSocketHandleImpl::sysCallResultToIoCallResult(const Api::SysCallSizeResult& re
            : Api::IoErrorPtr(new IoSocketError(result.errno_), IoSocketError::deleteIoError)));
 }
 
+Address::InstanceConstSharedPtr maybeGetDstAddressFromHeader(struct cmsghdr* cmsg,
```

Comment:
nit: can we pass `const cmsghdr&` here? Or at least a const pointer? Not sure if const will work with some of the old socket APIs.

Reply:
done